src/helpers/requests/fetch-request.js

```diff
@@ -0,0 +1,43 @@
+import MiniPromise from '../../ponyfills/minipromise.js';
+import { createSignatureHeader } from './create-signature-header.js';
+
+
+const URL_PATTERN = /^([a-z]+):\/\/([^/]+)(.*)/i;
+
+
+export default function fetchRequest(options) {
+  return MiniPromise.createPromise(function(resolve, reject) {
+    const parts = URL_PATTERN.exec(options.url);
+
+    if (!parts) {
+      throw new Error('Evolv: Invalid endpoint URL');
+    }
+
+    const headers = {
+      'Content-Type': options.encoding,
+      'Accept': 'application/json',
+      // eslint-disable-next-line no-undef
+      'Content-Length': Buffer.byteLength(options.payload)
+    };
+
+    if (options.signature) {
+      headers['Signature'] = createSignatureHeader(options.keyId, options.signature);
+    }
+
+    fetch(options.url, {
+      method: options.method,
+      headers: headers,
+    })
+      .then(function (res) {
+        if (res.status === 200) {
+          resolve(res.json());
+        } else if (res.status === 202) {
+          resolve();
+        } else {
+          const message = 'Evolv: Invalid status ' + res.statusCode + ' for response ' + res.statusMesssage;
```

Comment:
I see you changed `statusCode` to `status` for the other blocks. That should prolly apply here. IDE's also complaining about `statusMessage`. Is that the correct field?

Reply:
Hmm, that might have been Copilot's fault